[wx] Merge 'AddEditPropSheetDlg::UpdatePasswordTextCtrl' with 'ShowHideText' from wxUtilities to prevent redundant code

rafaelx opened the pull request:
- Directly/Dynamically changing the style flags of a password text control also improves the layout behavior of YubiCfgDlg.
- The ShowHideText code in wxUtilities was pretty much identical to that of AddEditPropSheetDlg::UpdatePasswordTextCtrl.

nobugshere commented:
I pulled from rafaelx:UpdatePasswordTextCtrl and verified the changes on a Mac, all looks good.

macOS M1max 14.4, wx 3.2.4, Xcode 15.3, with and without Rosetta.

igor2x commented:
@rafaelx, I really like you have taken time to write good comments inside source code.

For this:
"According to the documentation of wxTextCtrl the style flags "wxTE_PASSWORD" and "wxTE_READONLY" can be changed during runtime under wxGTK but not wxMSW."

Just a humble suggestion. Can you inside commend add URL like, where you have found this restriction. Someone may easily verify the claim, specially important over time (years) when restriction may be removed and in this case documentation probably also changes to indicate removal of restriction.

dskrvk commented on src/ui/wxWidgets/wxUtilities.h:
```diff
@@ -176,9 +176,24 @@ inline const wxChar* ToStr(bool b) {
+ *
+ * @see https://docs.wxwidgets.org/stable/classwx_text_ctrl.html
+ */
+void UpdatePasswordTextCtrl(wxSizer *sizer, wxTextCtrl* &textCtrl, const wxString text, wxTextCtrl* before, const int style);
```
Is it really necessary to put the comment in both the header and the source file?

rafaelx commented:
@nobugshere Great!
@igor2x I added a link in the notes section.
@dskrvk I removed the comment in the .cpp file because the .h file serves as an interface declaration.

Thank you everyone for the feedback.
